src/hotspot/share/opto/loopTransform.cpp

```diff
@@ -1471,19 +1437,19 @@ void PhaseIdealLoop::count_opaque_loop_nodes(Node* n, uint& init, uint& stride)
   wq.push(n);
   for (uint i = 0; i < wq.size(); i++) {
     Node* n = wq.at(i);
-    if (is_part_of_template_assertion_predicate_bool(n)) {
-      for (uint j = 1; j < n->req(); j++) {
-        Node* m = n->in(j);
-        if (m != nullptr) {
-          wq.push(m);
+    if (TemplateAssertionPredicateExpressionNode::valid_opcode(n)) {
```

Comment:
`valid_opcode()` also includes the `OpaqueLoop*` nodes which was not the  case for `is_part_of_template_assertion_predicate_bool()`. Therefore needed to adapt the code below to handle that.